eglot-jl.el

```diff
@@ -71,10 +76,31 @@ Otherwise returns nil"
 (cl-defmethod project-roots ((project (head julia)))
   (list (cdr project)))
 
+(defun eglot-jl--sysimage-args ()
+  "Returns a list of command-line switches suitable for sysimage loading.
+Return nil if no system image has been found or if system images
+have been disabled altogether."
+  (when eglot-jl-enable-sysimage
+    (let ((julia-version (with-temp-buffer
+                           (call-process eglot-jl-julia-command nil t nil
```

Comment:
Given the additional overhead of calling `julia`, I think my preference would be to default `eglot-jl-enable-sysimage` to nil. Obvious counter-argument is that there's no point in trying to avoid a call to julia when the language server itself takes 20+ seconds to start.

Reply:
About the overhead itself:

As it is written now, we're speaking of an overhead of a few hundreds of miliseconds (on my rather old system):
```
$ time julia -e 'print(VERSION)'
1.5.0-rc1.0

real    0m0,259s
user    0m0,328s
sys     0m0,203s
```

I had not thought about this until now, but this could be reduced by decreasing the compilation level:
```
$ time julia --compile=min -e 'print(VERSION)'
1.5.0-rc1.0

real    0m0,154s
user    0m0,077s
sys     0m0,070s
```

If we want to reduce this overhead further, the best option I can think of would be to use `julia --version`:
```
$ time julia --version
julia version 1.5.0-rc1

real    0m0,039s
user    0m0,020s
sys     0m0,001s
```
To be honest, the reason why I did not chose this option was because this involves additional parsing on the output but:
1. I was lazy and wanted to avoid doing it if possible, and
2. I did not know whether this format is normalized and consistently used across all julia versions.

One more word about this sub-issue regarding whether we should spawn a fully-fledged julia process or just use `julia --version`: the former is more flexible in that it allows doing more things in Julia, such as getting the versions of all packages, computing the correct file extension or other things like that. (More on that later in the answers to your other comments)

Reply:
On the broader discussion regarding the opt-in / opt-out mechanism:

Like you said, my initial rationale was that this "auto-detection" feature imposes an overhead that is completely negligible for users who don't opt in the feature, and acceptable to users who opt in.

I'd like to (if possible) avoid a situation where users who want to opt in would have to perform two manual actions: change a setting in their config file, and trigger the generation of the sysimage. Maybe an alternative mechanism would consist in having users opt-in by setting `eglot-jl-enable-sysimage` in their config file, and then `eglot-jl` would try and detect an existing, appropriate sysimage, use it if found or automatically compile one if none is found. What would you think of that?